WordPress/gutenberg
# Adds page list as child of submenu

## What?


Fixes #46413 - allows the page list block to be used inside submenus of a navigation block.

## Why?


Because now that Page list is filterable on parent it can be used to create self updating sections of the navigation.

## How?


- enables page list as a child of navigation link and navigation submenu
- adds a transform from navigation link to page list
- adds an attribute to page list that specifies whether the block is nested in a submenu block or not
- fixes some problems of page list with styling
  - in the editor the color props from the navigation block were not passed down to all nested inner blocks but only to the 1st level
  - on the front end the color props from the navigation block were only applied to the 1st level of nested inner blocks
- copies over one of the changes in #46416 that aligns the off canvas editor with the list view in how it receives the list of blocks, changing from passing the inner blocks of the selected navigation block to a tree of client ids.

## Testing Instructions

1. make sure to have some pages that are nested in a hierarchy
2. in a new post
3. add a navigation block
4. in the navihation block add a link and select it
5. click the transform menu in the navigation link's toolbar
6. click page list
7. **there should be a page list with all the pages now in the submenu**
8. select the parent navigation block
9. in the inspector in the color panel set colors for background and text for overlays
10. Save
11. **In the editor the colors are respected**
12. **On the front end the page list renders well in the menu and the colors are respected**

### Testing Instructions for Keyboard


N/A

## Screenshots or screencast 


https://user-images.githubusercontent.com/107534/206705918-f70af1a0-f02c-422f-8c83-e15ce6521d45.mp4
